swap_meet/vendor.py

```diff
@@ -1,2 +1,77 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory= None):
+        inventory = inventory if inventory is not None else []
+        self.inventory = inventory
+
+
+    def add(self, item): 
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item_to_remove):
+        if not self.inventory:
+            return False
+        elif item_to_remove in self.inventory:
+            item_index = self.inventory.index(item_to_remove)
+            popped_item = self.inventory.pop(item_index)
+            return popped_item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        category_list = []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)   
+        return category_list
+
+    def swap_items(self, Vendor, my_item, their_item):
```

Comment:
Here you passed in the class `Vendor` instead of a variable `vendor`, instead of 

In Python we don't need to specify types so the `Vendor` class can just be the variable `vendor` (with a lowercase v). If you _really_ want to ensure that the parameter actually is a Vendor, you can add an assert isinstance(vendor, Vendor) or if not isinstance(vendor, Vendor): raise TypeError

But that's not recommended in Python. 

The point of "duck typing" is that you write a function that takes "anything that acts like a `Vendor` instance" rather than a function that takes "a `Vendor` instance".

Here's a resource on duck typing if you're interested in reading more: https://realpython.com/lessons/duck-typing/#:~:text=Duck%20typing%20is%20a%20concept,a%20given%20method%20or%20attribute.